### PR TITLE
Add a blog post detailing uses for skipExecutingMigrations

### PR DESCRIPTION
**Pull request by Lyeeedar**

### CODE REVIEW

File: _posts/2020-10-16-skipExecutingMigrations.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+Environments:
+dev: 1,2,3,3.1
+test: 1,2,3,3.1
+prod: 1,2,3 -- error!
+```
+
+However, when you then try to `migrate` on prod, the migration would fail, as the changes applied by the hotfix already exist!
+
+This is where `skipExecutingMigrations` comes in. On prod instead of running a normal `migrate`, instead run `migrate -skipExecutingMigrations=true`. Now Flyway will not actually try to execute the contents of the migration on this environment, instead just updating the schema history to say its been applied. Now all the environments are back in sync, and development can continue.
```
Comment (juliahayward) [inline _posts/2020-10-16-skipExecutingMigrations.md]: *it's* been

File: _posts/2020-10-16-skipExecutingMigrations.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+Now a developer can check out this project, and apply the changes to a local dev database. Then they can open the database in an IDE of their choice and make edits to it. Once they are satisfied with the changes they can use a tool to generate a migration script from their changes, and save it as a new migration on disk.
+
+```
+V1__create_tables.sql
+V2__modify_tables.sql
+V3__create_views.sql
+V4__new_changes.sql
+```
+
+Migration `V4` can't be applied to the developers own database as the changes already exist. In this case the developer can then run `migrate -skipExecutingMigrations=true` to mark the migration as applied locally. The migration can then be committed to VCS, and any other developer can update and apply the migration without a problem to their own dev instances.
```
Comment (juliahayward) [inline _posts/2020-10-16-skipExecutingMigrations.md]: *developer's*

File: _posts/2020-10-16-skipExecutingMigrations.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+
+## Deferred migration execution
+
+In more complex projects, it is sometimes desirable to have more control over *when* a migration is executed (such as during down time), out of order from the other migrations. This could be because the migration takes a long time to execute, its deemed too risky by the dba and wants to be skipped for now, or many other reasons.
```
Comment (juliahayward) [inline _posts/2020-10-16-skipExecutingMigrations.md]: *is* deemed? Or *it's* ?

File: _posts/2020-10-16-skipExecutingMigrations.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+U13__undo_a_very_slow_operation.sql
+U14__delete_more_things.sql
+U15__unshuffle_things.sql
+U16__delete_things.sql
+```
+
+In this example, `V13` takes a very long time to execute, and the DBA decided that the best time to execute the migration is during the quiet period at the weekend. However, the rest of the migrations still need to be applied immediately.
+
+Before Flyway 7.0.0, the only answer would be to delete the migration from disk (and VCS, just incase an accidental `migrate` applies it before the DBA is ready), and reintroduce it once it is desirable again to apply it. Pretty fiddly!
```
Comment (juliahayward) [inline _posts/2020-10-16-skipExecutingMigrations.md]: *in case*

File: _posts/2020-10-16-skipExecutingMigrations.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+Now, when the time comes to actually run the migration for real, the DBA can selectively undo the skipped migrations using [cherryPick](/documentation/configuration/parameters/cherryPick) and `skipExecutingMigrations`, marking it as pending once again without actually running any of the undo script:
+
+`flyway undo -cherryPick=13 -skipExecutingMigrations=true`
+
+Now a `migrate -outOfOrder=true` will execute the `V13` migration that was delayed.
+
+
+## Intermediate baseline
+
+As a project grows in size, constructing the databases used in various temporary environments (such as dev or test) becomes harder to manage via a normal `migrate`. Instead of applying a vast number of migrations to setup the database, it may because more desirable to create the database via other means. This could be a new migration that does the same thing all the previous migrations did in a more efficient way, restoring from some form of image or backup, or something else. No matter what method is chosen, the problem of what to do with the existing migration files remains.
```
Comment (juliahayward) [inline _posts/2020-10-16-skipExecutingMigrations.md]: *become* instead of "because"?

Comment (DoodleBobBuffPants) [inline _posts/2020-10-16-skipExecutingMigrations.md]: It sounds like `it may because` should be `it may become`?

**Review by juliahayward — COMMENTED** [review]
Great! A few bits of pedantry but the content is really good giving plenty of use cases.

**Review by MikielAgutu — APPROVED** [review]

**Pull request merged**